dom.go

```diff
@@ -872,19 +870,20 @@ func extractHTML(e ComponentOrHTML) *HTML {
 	case *HTML:
 		return v
 	case Component:
-		return v.Context().prevRender
+		return extractHTML(v.Context().prevRender)
 	default:
 		panic("vecty: encountered invalid ComponentOrHTML " + reflect.TypeOf(e).String())
 	}
 }
 
-// sameType returns whether first and second components are of the same type
-func sameType(first, second Component) bool {
+// sameType returns whether first and second ComponentOrHTML are of the same
+// type.
+func sameType(first, second ComponentOrHTML) bool {
```

Comment:
Since this now operates on `ComponentOrHTML`, it might be nice to add a new docstring below the existing one like:

```Go
// Two *HTML are always of the same type, whereas two Component have their
// concrete underlying types compared.
```

WDYT?

Reply:
Is it sufficient just to say:

```go
// sameType returns whether first and second ComponentOrHTML are of the same
// underlying type.
```

Otherwise we can describe various typical comparisons, *HTML to *HTML, *HTML to Component, to different components.  I don't feel strongly about it either way.

Reply:
Yep that's sufficient.